### ISSUE TITLE
component [spreadsheetLibrary.Spreadsheet] has no function with name [getActiveSheetIndex]

### ISSUE THREAD

**Issue by xremix** [cfsimplicity/lucee-spreadsheet#89]
I have some issues with several functions of `workbook` that can't be found, like the `getActiveSheetIndex()` or the `getLastRowNum()` function. The errors do throw inside the library. I'm not sure if this might be related to my code set up, or if this is an bug in you library. That's why I open an issue and ask for help here. Hopefully you can help me, as I really like the work you did with the spreadsheet library.

![bildschirmfoto 2016-08-08 um 11 15 26](https://cloud.githubusercontent.com/assets/9824294/17475155/e26780a0-5d59-11e6-8c53-8fbe07639d14.png)

I'm using the version `0.7.8` of the library and run it under CF 10 and Lucee `4.5.3`.

**Comment by cfsimplicity** [comment]
Can you post the code you are using to call the library when it throws these errors?

**Comment by cfsimplicity** [comment]
The error message suggests you're passing in the library instance itself rather than a workbook object to whichever public library method you're using.

**Comment by xremix** [comment]
For sure. 
I took a further look and tried to run the sample from your README, and I was having the same issue with this sample:
```
<cfscript>
spreadsheet=New spreadsheetLibrary.Spreadsheet();
data=QueryNew( "First,Last","VarChar,VarChar",[ [ "Susi","Sorglos" ],[ "Frumpo","McNugget" ] ] );
workbook=spreadsheet.new();
spreadsheet.addRows( workbook,data );
</cfscript>
```
This is the code I'm running:
```
spreadsheet=New spreadsheetLibrary.Spreadsheet();
queryData = spreadsheet.read(src="#theFile#", format="query", headerrow=1);
...
// Setting query data
...
spreadsheet.addRows(workbook=spreadsheet, data="#queryData#")
```
And I guess you are right, that I passed the spreadsheet instead of the workbook, but I tried the same thing like in your sample, by creating a new workbook instance and passing it to the `addRows` method, but I experienced no difference.

**Comment by cfsimplicity** [comment]
Can you post your revised code where you create the new workbook and pass it into `addRows()` as per the sample?

**Comment by xremix** [comment]
My revised code looks like
```
spreadsheet=New spreadsheetLibrary.Spreadsheet();
queryData = spreadsheet.read(src="#theFile#", format="query", headerrow=1);
...
// Generate Query data here
...
workbook=spreadsheet.new();
spreadsheet.addRows(workbook=workbook, data="#queryData#")
```

And like I said I actually get the same error with your sample:
````
sheet=New spreadsheetLibrary.Spreadsheet();
data=QueryNew( "First,Last","VarChar,VarChar",[ [ "Susi","Sorglos" ],[ "Frumpo","McNugget" ] ] );
workbook=sheet.new();
sheet.addRows( workbook,data );
```

Thanks already a lot for taking  a look and your time in advance

**Comment by cfsimplicity** [comment]
Strange. Just tested the sample code again and it's working fine for me. If you're getting the same error as before then there must still be something wrong with the `workbook` argument you are passing into `addRows()`. Try dumping that variable before you pass it into that function, i.e.
```
spreadsheet=New spreadsheetLibrary.Spreadsheet();
queryData = spreadsheet.read(src="#theFile#", format="query", headerrow=1);
// Generate Query data here
workbook=spreadsheet.new();
dump( workbook );
abort;
spreadsheet.addRows(workbook=workbook, data="#queryData#")
```
What do you get?

**Comment by xremix** [comment]
I agree with you.
I get a workbook that, for me, looks ok:

![bildschirmfoto 2016-08-08 um 12 55 58](https://cloud.githubusercontent.com/assets/9824294/17477779/74cded64-5d67-11e6-88d8-ac4a42420b44.png)

Also I can see an existing function `getActiveSheetIndex()` there.

**Comment by cfsimplicity** [comment]
Try aborting aborting immediately after the `addRows()` call.
```
spreadsheet=New spreadsheetLibrary.Spreadsheet();
queryData = spreadsheet.read(src="#theFile#", format="query", headerrow=1);
// Generate Query data here
workbook=spreadsheet.new();
spreadsheet.addRows(workbook=workbook, data="#queryData#");
abort;
```
Do you still see the error?

**Comment by xremix** [comment]
I tried to debug this and came to  the code line 1598 in the Library where the function `private any function getActiveSheet( required workbook )` starts and dumped the workbook here as well, and it seems to be fine at that point. Anyway the function that the error was complaining about wasn't there. For some reason, after playing around and after some back and forth this works now. The really weird thing is that the code that I call, as well as the code of the library now has the same status like at the point where I shared my code (and yes, I'm sure it is the same :) ). I guess that my issue might have something todo with the caching of my server, wich shouldn't be the case as well as I restarted the server several times and cleared the cache of the browser as well..
Anyways I'm closing this now and really appreciate your help.

I'll reopen this if I figure out more or get the error once again.

Thanks a lot for your time and help!
